framework/packages/abstract-sdk/src/apis/feegrant.rs

```diff
@@ -0,0 +1,348 @@
+//! # Fee Granter
+//! This module provides functionality to interact with the feegrant module of Cosmos.
+//! It allows for granting fee expenditure rights to other accounts.
+
+use std::time::Duration;
+
+use cosmos_sdk_proto::traits::TypeUrl;
+use cosmos_sdk_proto::{cosmos::base, cosmos::feegrant, traits::Message, Any};
+use cosmwasm_std::{Addr, Binary, Coin, CosmosMsg, Timestamp};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::cell::RefCell;
+
+use crate::features::AccountIdentification;
+use crate::AbstractSdkResult;
+
+/// An interface to the CosmosSDK FeeGrant module which allows for granting fee expenditure rights.
+pub trait GrantInterface: AccountIdentification {
+    /// API for accessing the Cosmos SDK FeeGrant module.
+    /// The **granter** is the address of the user granting an allowance of their funds.
+    /// By default, it is the proxy address of the Account.
+
+    /// ```
+    /// use abstract_sdk::prelude::*;
+    /// # use cosmwasm_std::testing::mock_dependencies;
+    /// # use abstract_sdk::mock_module::MockModule;
+    /// # let module = MockModule::new();
+    /// # let deps = mock_dependencies();
+
+    /// let grant: FeeGranter = module.fee_granter(deps.as_ref(), None);
+    /// ```
+    fn fee_granter<'a>(
+        &'a self,
+        deps: cosmwasm_std::Deps<'a>,
+        granter: Option<Addr>,
+    ) -> FeeGranter<Self> {
+        FeeGranter {
+            base: self,
+            deps,
+            granter: RefCell::new(granter),
+        }
+    }
+}
+
+impl<T> GrantInterface for T where T: AccountIdentification {}
+
+/// This struct provides methods to grant fee allowances and interact with the feegrant module.
+///
+/// # Example
+/// ```
+/// use abstract_sdk::prelude::*;
+/// # use cosmwasm_std::testing::mock_dependencies;
+/// # use abstract_sdk::mock_module::MockModule;
+/// # let module = MockModule::new();
+///
+/// let grant: FeeGranter  = module.fee_granter();
+/// ```
+/// */
+#[derive(Clone)]
+pub struct FeeGranter<'a, T: GrantInterface> {
+    base: &'a T,
+    deps: cosmwasm_std::Deps<'a>,
+    // We use a RefCell here to allow for the granter to be set after the FeeGranter is created
+    granter: RefCell<Option<Addr>>,
```

Comment:
This should just be owned type. The struct can be `mut` if the user wants to set the granter.

I.e. change `fn granter(&self, ...)` to `fn granter(&mut self, ...)`